### PR TITLE
FE | Complete User Story 7

### PR DESCRIPTION
**Pull request by HuyPhan2025**
# Changes 
[ ] Bug fix
[x] New feature
[ ] This change requires a documentation update

## List of Changes: 
- <p>This PR is related to this issue: #19</p>
- Create random act facade and spec
- Create random act service and spec
- Create random act poro 
- Create random act controller
- Add random act json to fixture file
- Create random act index html
- Create good deeds new html
- Create good deeds login html
- Update application controller with `current_user` method
- Update good deeds controller `new` method

# Self Review
[ ] I have performed a self-review of my code.
[ ] I have added thorough tests.
[x] My changes generate no new warnings.
[x] I have linted my code locally before submission.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class ApplicationController < ActionController::Base
+  helper_method :current_user
+
+  def current_user
```
Comment (KaraJoHo) [inline app/controllers/application_controller.rb:4]: Nice! Helper methods are the best

**Review by KaraJoHo — APPROVED** [review]
This is great. Login functionality and error messages look neat! Lots of good testing

File: app/services/random_act_service.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class RandomActService
+  def self.conn
+    Faraday.new(url: "https://localhost:3000")
```
Comment (MelTravelz) [inline app/services/random_act_service.rb:3]: This looks great!!! As a refactor we can consider taking the end of the URL from line 12 and connecting it to the URL in line 3. On the BE this API call will never change and so what it delivers to the FE will probably also never change. : )

File: app/controllers/good_deeds_controller.rb [hunk 3]
```diff
@@ -3,7 +3,12 @@ def index
...
+  def new
+    return unless current_user.nil?
+
+    redirect_to login_path
+    flash[:alert] = "You must be logged in to create a new good deed"
   end
-end
\ No newline at end of file
+
+  def login; end
```
Comment (hamouj) [inline app/controllers/good_deeds_controller.rb:13]: Moving forward, let's move the login and logout functionality to a sessions controller.

File: config/routes.rb [hunk 4]
```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
+  get "/", to: "good_deeds#index"
```
Comment (MelTravelz) [inline config/routes.rb:2]: As a refactor we could consider using references as the BE app is using.

**Pull request merged**